src/output_data.py

```diff
@@ -134,31 +134,63 @@ def output_data_quality_metrics(df, file_name, path_prefix, skip_outputs):
         )
 
 
-def output_plant_data(df, path_prefix, resolution, skip_outputs):
+def _identify_mixed_method_plants(eia923_allocated):
+    """
+    Using eia923 after hourly method assignment,
+    identify plants whose hourly output will not be complete
+    (ie, some subplants are CEMS, partial_cems, or partial_cems_plant,
+    while others are EIA)
+    Helper function for output_plant_data.
+    """
+
+    def split_hourly(hourlies):
+        # if EIA is a method and there is more than one method,
+        # then the hourly plant-level data will not contain all subplants
+        return ("eia" in hourlies.to_list()) & (hourlies.nunique() > 1)
+
+    methods = (
+        eia923_allocated.groupby(["plant_id_eia", "report_date"])
+        .hourly_data_source.apply(split_hourly)
+        .reset_index()
+    )
+    methods = methods[methods.hourly_data_source]
+    return methods.plant_id_eia.unique()
+
+
+def output_plant_data(df, path_prefix, resolution, skip_outputs, eia923=np.NaN):
     """
     Helper function for plant-level output.
     Output for each time granularity, and output separately for real and shaped plants
+    `df` contains all plant-level data, both CEMS and synthetic.
+    `eia923` is used to identify plants whose hourly data comes from different sources; none
+        of its data is written or modified here (optional, only needed for hourly)
 
     Note: plant-level does not include rates, so all aggregation is summation.
     """
     if not skip_outputs:
         if resolution == "hourly":
             # output hourly data
             # Separately save real and aggregate plants
+            # Separately save complete plant CEMS and subplant CEMS
+            partial_plants = _identify_mixed_method_plants(eia923)
```

Comment:
Is this still necessary if we are just using `results/plant_data/plant_metadata.csv` to indicate when a plant is split up?